src/main/java/edu/umich/its/cpm/MigrationController.java

```diff
@@ -1,32 +1,196 @@
+			{
+				log.error(requestUrl + e.getMessage());
+			}
+        }
+		return rv;
+    }
 
+	/**
+	 * login into CTools and become user with sessionId
+	 */
+	private String login_becomeuser() {
+		// return the session id after login
+		String sessionId = "";
+		
+		// here is the CTools integration prior to CoSign integration ( read session user information from configuration file)
+    	// 1. create a session based on user id and password
+    	RestTemplate restTemplate = new RestTemplate();
+    	// the url should be in the format of "https://server/direct/session?_username=USERNAME&_password=PASSWORD"
+    	String requestUrl = env.getProperty("ctools.direct.url") + "session?_username=" + env.getProperty("username") + "&_password=" + env.getProperty("password");
+    	ResponseEntity<String> response = restTemplate.postForEntity(requestUrl, null, String.class);
+        HttpStatus status = response.getStatusCode();
+        
+       if (status.value() != 201)
+        {
+        	// return error if a new CTools session could not be created using username and password provided
+        	log.info("Wrong user id or password. Cannot login to CTools " + env.getProperty("ctools.direct.url"));
+        }
+        else
+        {
+        	// get the session id
+            sessionId = response.getBody();
+            log.info("successfully logged in as user " + env.getProperty("username") + " with sessionId = " + sessionId);
+            
+        	// 2. become the user
+        	// retrieve from the configuration file for now; Shoudl get from the REMOTE_USER setting after CoSign integration
+        	restTemplate = new RestTemplate();
+        	// the url should be in the format of "https://server/direct/session/SESSION_ID.json"
+        	requestUrl = env.getProperty("ctools.direct.url") + "session/becomeuser/" + env.getProperty("become.username")+".json?_sessionId=" + sessionId;
+        	log.info(requestUrl);
+
+        	try
+        	{
+        		String resultString = restTemplate.getForObject(requestUrl, String.class, sessionId); 
+        		log.info(resultString);
+        	}
+        	catch (RestClientException e)
+        	{
+        		log.error(requestUrl + e.getMessage());
+        		
+        		// nullify sessionId if become user call is not successful
+        		sessionId = null;
+        	}
+        }
+        return sessionId;
+    }
+
+    /**
+     * get all migration records
+     * @return
+     */
     @RequestMapping("/migrations")
     public String migrations() {
-    	Gson gson = new Gson();
-    	return gson.toJson(repository.findAll());
+    	return JSONObject.valueToString(repository.findAll());
     }
     
+    /**
+     * get a specific migration record
+     * @param migration_id
+     * @return
+     */
+    @RequestMapping("/migrations/{migration_id}")
+    public String migrations(@PathVariable("migration_id") Integer migration_id) {
+    	return JSONObject.valueToString(repository.findOne(migration_id));
```

Comment:
Code updated. Now if the migration_id is invalid, "no matching record for /migrations/MIGRATION_ID" is returned in JSON string.